client/dev/navbar.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+class NavLink extends React.Component{
+  render(){
+    return (
+      <a href={this.props.url}>{this.props.linktext}</a>
+    )
+  }
+}
+
+class NavItem extends React.Component {
+  makeLink(){
+    return <NavLink url={this.props.url} linktext={this.props.linktext} />;
+  }
+  
+  makeSubMenu(){
+    return <NavBar items={this.props.submenu} />;
+  }
+  
+  makeNavContent(){
+    var content = [this.makeLink()];
+    if(this.props.submenu){
+      content.push(this.makeSubMenu());
```

Comment:
Be cautious about using methods that mutate data structures while using React. This is fine because the new Array is created in this function, so you are not mutating state, but something to keep in mind.